slides/12_Remix.html

```diff
@@ -24,6 +24,727 @@
+            <pre><code class="hljs tsx" data-trim data-line-numbers="|8,9,14,15,20,21|2,12,13,16-19|16|1,4|"><script type="text/template">
+import type { LinksFunction } from "@remix-run/node";
+import { Links, LiveReload, Meta, … } from "@remix-run/react";
+
+export const links: LinksFunction = () => […];
+
+export default function App() {
+  return (
+    <html lang="en">
+      <head>
+        <meta charSet="utf-8" />
+        <meta name="viewport" content="width=device-width,initial-scale=1" />
+        <Meta />
+        <Links />
+      </head>
+      <body>
+        <Outlet />
+        <ScrollRestoration />
+        <Scripts />
+        <LiveReload />
+      </body>
+    </html>
+  );
+}
+          </script></code></pre>
+            <aside class="notes">
+              Some notes:
+              <ol>
+                <li>Describes complete document (&lt;html&gt; element)</li>
+                <li>
+                  Special Remix components for app setup (again, full control)
+                </li>
+                <li>
+                  Router outlet (same as in React Router): your page is rendered
+                  here
+                </li>
+                <li>
+                  <em>links</em> export: e.g. global stylesheets. Conventions
+                  explained in next chapter
+                </li>
+              </ol>
+            </aside>
+          </section>
+        </section>
+
+        <section>
+          <section>
+            <h3>NPM Scripts</h3>
+            <p><span class="code">npm run dev</span></p>
+            <p>Starts the app in dev mode with hot reload</p>
+          </section>
+          <section>
+            <p><span class="code">npm run build</span></p>
+            <p>
+              Creates a production build of your app (output in
+              <span class="code">build</span> and
+              <span class="code">public/build</span> folders)
+            </p>
+          </section>
+          <section>
+            <p><span class="code">npm start</span></p>
+            <p>
+              Starts Node.js server in <span class="code">build</span> folder
+            </p>
+            <div class="tip fragment">
+              For deployment to common platforms consider using a template (see
+              <span class="code">README.md</span>)
+            </div>
+          </section>
+          <section>
+            <p><span class="code">npm run typecheck</span></p>
+            <p>
+              Runs TypeScript compiler to ensure type-safety in build pipeline
+              (<em>not</em>
+              done automatically by Remix compiler)
+            </p>
+          </section>
+        </section>
+
+        <section class="subtitle-slide">
+          <h1>Routing</h1>
+        </section>
+
+        <section>
+          <section>
+            <h3>Intro to Route File Naming</h3>
+          </section>
+          <section>
+            <h4>General conventions</h4>
```

Comment:
**Suggestion:** Maybe mention that the route definition in remix is done through a file path convention instead of a router definition like in barebone react router (even though remix uses the same library under the hood).